torch/csrc/distributed/rpc/torchscript_functions.cpp

```diff
@@ -20,13 +21,21 @@ c10::intrusive_ptr<c10::ivalue::Future> rpcTorchscript(
     std::vector<c10::IValue>& stack,
     const float rpcTimeoutSeconds,
     const bool isAsyncExecution) {
-  if (torch::autograd::profiler::profilerEnabled() &&
+  // This dummy tensor holds an at::RecordFunction when profiling is enabled.
+  // This is because at::RecordFunction is not yet registered as a TorchScript
+  // custom class (https://github.com/pytorch/pytorch/issues/35026)
+  at::Tensor handle = at::zeros(1);
+  auto shouldProfile = torch::autograd::profiler::profilerEnabled() &&
       !torch::distributed::rpc::RemoteProfilerManager::getInstance()
-           .isCurrentKeySet()) {
+           .isCurrentKeySet();
```

Comment:
Is there a unit test that exercises the `isCurrentKeySet` condition?

Reply:
Yes, since this code path is triggered both by `rpc_async` being called with a `ScriptFunction` in normal python mode, and also `rpc_async` call from a TorchScript function. In the former case, `isCurrentKeySet` is true since we set that in python (`rpc/api.py`). So any of the tests such as `test_profiler_with_script_async_rpc` will test the case where it's true, and the case where it's false is tested in the test added to this diff.